roles/cdi/tasks/provision.yml

```diff
@@ -0,0 +1,26 @@
+---
+# General State of World Checks
+# thought I might need this in case had to do OC specific things
+# but seems I do not, but leaving for now for future use
+- name: Determine Environment
+  shell: "oc version> /dev/null 2>&1; if [ \"$?\" -eq 127 ]; then echo 'kubectl'; else echo 'oc'; fi"
+  register: cli
+
+- name: Check if CDI is already running
+  shell: "kubectl get deployment -n {{ cdi_namespace }} | grep -w 'cdi-deployment' | awk '{ print $1 }'"
+  register: cdi_running
+  ignore_errors: yes
+
+# General CDI Deployment
+- name: Check if namespace {{ cdi_namespace }} exists
+  shell: kubectl get ns | grep -w {{ cdi_namespace }} | awk '{ print $1 }'
+  register: ns
+
+- name: Create {{ cdi_namespace }} namespace
+  shell: kubectl create namespace {{ cdi_namespace }}
+  when: ns.stdout != cdi_namespace
+
+- name: Create CDI deployment
+  command: "kubectl create -f https://raw.githubusercontent.com/kubevirt/containerized-data-importer/master/manifests/cdi-controller-deployment.yaml -n {{ cdi_namespace }}"
```

Comment:
I always like external dependencies to be tagged so a change in the the master branch of cdi won't break this repo.  Kubevirt handles this by creating tagged releases including the manifests.  If you go that route, be sure to make the tag you use a variable at the top-level of the role so we can adjust it easily.